src/validators/mod.rs

```diff
@@ -50,7 +51,17 @@ impl SchemaValidator {
             }
         };
         let slots = slots_builder.into_slots()?;
-        Ok(Self { validator, slots })
+        Ok(Self {
+            validator,
+            slots,
+            schema: Into::<Py<PyAny>>::into(schema),
+        })
+    }
+
+    fn __reduce__<'py>(&self, py: Python<'py>) -> PyResult<&'py PyTuple> {
+        let args = PyTuple::new(py, [self.schema.as_ref(py)]);
+        let cls = Py::new(py, self.to_owned())?.getattr(py, "__class__")?;
```

Comment:
Is there a better way of getting the current class?

Reply:
What about
```suggestion
        let cls = self.to_object().getattr(py, "__class__")?;
```
? No idea if that would work.

If not I'm happy with this.

Reply:
did not work 😞

Reply:
🤷